projects/packages/blocks/src/class-blocks.php

```diff
@@ -99,7 +117,52 @@ function () use ( $feature_name, $method_name ) {
 			}
 		}
 
-		return register_block_type( $slug, $args );
+		return register_block_type( $block_type, $args );
+	}
+
+	/**
+	 * Read block metadata from a block.json file.
+	 *
+	 * @param string $filename The path to the block.json file or its directory.
+	 *
+	 * @return array The block metadata.
+	 */
+	public static function get_block_metadata_from_file( $filename ) {
+		$metadata = array();
+		$needle   = '/block.json';
+		$filename = $needle === substr( $filename, -strlen( $needle ) ) ? $filename : realpath( $filename . $needle );
+
+		if ( file_exists( $filename ) ) {
+			try {
+				$metadata = wp_json_file_decode( $filename, array( 'associative' => true ) );
+			} catch ( Exception $e ) {
```

Comment:
I'm not sure if this is really necessary, but wanted to suggest it anyway:

```suggestion
			} catch ( \Exception $e ) {
```